doc/development/deprecations.rst

```diff
@@ -51,6 +51,11 @@ Pending deprecations
         some_qfunc(params)
         return qml.expval(Hamiltonian)
 
+* The public methods of `DefaultQubit` that aren't present in `DefaultQubit2` now raise a deprecation
+  warning if called directly by users. The warning is not raised in normal workflows.
+
+  - Deprecated in v0.31
+
```

Comment:
I wonder if its worth adding some more context here, along the lines of 

```
We are switching to a new device interface. In this process, simulation implementation details are abstracted away from the device class itself and provided by composition, rather than inheritance. Therefore, many public and private methods from `DefaultQubit` will no longer exist, though its behaviour in a workflow will remain the same. See :class:`pennylane.devices.experimental.Device` for more information on what the new interface will look like.

To directly work with the components of a Python state vector simulation, please consult the `devices.qubit` submodule.
```

I wonder if we should add a table of contents for `devices.qubit` on the `devices` "Internals" page.